### PR TITLE
Readthedocs

### PR DESCRIPTION
**Pull request by bentheiii**
https://heksher.readthedocs.io/en/readthedocs/

### CODE REVIEW

File: docs/setting_aliases.rst [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+Each setting has a list of aliases. Any setting can be referred to by any of its aliases. When a setting is renamed,
+it's old name is added to the list of aliases.
+
+For example:
+
+#. A new setting with name "foo" is declared. The setting is now accessible with its canonical name "foo".
+#. The setting is renamed to "bar". The setting is now accessible with either name "bar" or "foo". Its canonical name is
+   "bar".
+#. The setting is renamed to "baz". The setting is now accessible with "baz", "bar" or "foo". Its canonical name is
+   "baz".
```
Comment (omerhazan) [inline docs/setting_aliases.rst:16]: As a reference to yesterday conversation, what will happen if:
* Service A version 1.0 is declaring a setting named foo
* Service A version 1.1 is renaming foo to bar
* Service A version 1.2 is renaming bar to baz - but version 1.1 of the service is running in some customers

This is not relevant to the current CR :)

**Review by omerhazan — CHANGES_REQUESTED** [review]
Also add a link to the doc in the README

File: docs/api.rst [hunk 2]
```diff
@@ -0,0 +1,426 @@
+API
+============
+
+Here we detail all the HTTP API endpoints, their payloads and the expected responses.
+All endpoints return a JSON object, and those that expect a body in the request, expect it as a
```
[the remark this answers is not in the document]
Comment (bentheiii) [reply docs/api.rst:5]: All of is qualified, the apis are not qualified
https://getproofed.com/writing-tips/word-choice-all-of/#:~:text=Use%20all%20of%20when%20the,things%20or%20an%20uncountable%20noun.

File: docs/api.rst [hunk 3]
```diff
@@ -0,0 +1,426 @@
+API
```
[the remark this answers is not in the document]
Comment (bentheiii) [reply docs/api.rst:1]: boy did I try but rst and redoc were not playing nice

File: docs/api.rst [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+If a rule does not exists to that setting and feature_values, returns a 404 response.
+
+Otherwise, the response schema is:
+
+* rule_id: The id of the rule that was found.
+
+PATCH /api/v1/rules/<rule_id>
+******************************
+
+Change a rule's value.
```
[the remark this answers is not in the document]
Comment (bentheiii) [reply docs/api.rst:135]: you can, read ahead

File: docs/api.rst [hunk 5]
```diff
@@ -0,0 +1,426 @@
...
+
+Heksher will attempt to consolidate the following differences, if they exist:
+* If the declaration contains configurable_features that do not exist in the setting, they will be added to the setting.
+
+    * If the declaration does not contains configurable_features that do exist in the setting, they will **not** be removed
+      from the setting, the complete value will be indicated in the response.
+
+* If the type declared is a supertype of the actual type, the actual type will be updated to the declared type.
+
+    * If the type declared is a subtype of the actual type, the complete value will be indicated in the response.
```
[the remark this answers is not in the document]
Comment (bentheiii) [reply docs/api.rst:288]: it's intentional

File: docs/index.rst [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+.. toctree::
+    :maxdepth: 2
+    :caption: Contents:
+
+    introduction
+    concepts
+    running
+    setting_types
+    api
+    setting_aliases
```
[the remark this answers is not in the document]
Comment (bentheiii) [reply docs/index.rst:13]: I disagree, setting aliases are a more niche subject then general usage, reader can skip ahead if they're curious about alaising

**Review by omerhazan — APPROVED** [review]

**Pull request merged**